### ISSUE TITLE
Malformed version number string trusty64

### ISSUE THREAD

**Issue by philippgille** [hashicorp/vagrant#10663]
### Vagrant version

Vagrant 2.2.3

### Host operating system

Ubuntu 14.04

### Expected behavior

A list of the boxes on my machine should be shown.

### Actual behavior

The following error occurs:

```
/opt/vagrant/embedded/lib/ruby/2.4.0/rubygems/version.rb:208:in `initialize': Malformed version number string trusty64 (ArgumentError)
	from /opt/vagrant/embedded/lib/ruby/2.4.0/rubygems/version.rb:199:in `new'
	from /opt/vagrant/embedded/lib/ruby/2.4.0/rubygems/version.rb:199:in `new'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/box_collection.rb:251:in `block in all'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/box_collection.rb:250:in `each'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/box_collection.rb:250:in `sort_by'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/box_collection.rb:250:in `sort_by!'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/box_collection.rb:250:in `all'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/plugins/commands/box/command/list.rb:25:in `execute'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/plugins/commands/box/command/root.rb:66:in `execute'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/cli.rb:58:in `execute'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/lib/vagrant/environment.rb:291:in `cli'
	from /opt/vagrant/embedded/gems/2.2.3/gems/vagrant-2.2.3/bin/vagrant:182:in `<main>'
```

### Steps to reproduce

1. `vagrant box list`

### References

There are 14 related issues, you can find them with this search:

- https://github.com/hashicorp/vagrant/search?o=desc&q="Malformed+version+number+string"&s=created&type=Issues

I skimmed through some of them and didn't find a solution for my problem.

**Comment by briancain** [comment]
@philippgille - Are you using a custom box? It looks like whatever box you are using has listed the version number as `trusty64`, which is not a proper version for Vagrant boxes.

**Comment by philippgille** [comment]
@briancain: I never created my own boxes and I don't think my colleagues did either. But a while ago (maybe a year or so) when I used an older version of Vagrant (< 2.x) I had issues automatically downloading boxes when executing `vagrant up` in a directory with a Vagrantfile, so I had to manually add a box from Atlas, if I remember correctly.

How can I get rid of the `trusty64` box? `vagrant box prune` doesn't work. I wanted to try `vagrant box remove` but I don't know which boxes I have due to the issue at hand.

Is there a way to manually remove it, with vagrant-unrelated file operations (like `rm -rf /path/to/box` or so)? I didn't find anything with a quick Google search.

**Comment by briancain** [comment]
@philippgille - boxes are stored in your vagrant "home" directory, so if you're on a linuxy system it should be at `~/.vagrant.d/boxes`. You can find the box with the issue there!

**Comment by philippgille** [comment]
This was the list of boxes in my `~/.vagrant.d/boxes` directory:
```
bento-VAGRANTSLASH-ubuntu-16.04/
centos-VAGRANTSLASH-7/
precise64/
ubuntu/
ubuntu-VAGRANTSLASH-bionic64/
ubuntu-VAGRANTSLASH-trusty64/
ubuntu-VAGRANTSLASH-xenial64/
```

I removed the trusty box with `rm -rf ~/.vagrant.d/boxes/ubuntu-VAGRANTSLASH-trusty64`, but it didn't help. Then I removed the `ubuntu` directory, which had a subdirectory called `trusty64`. That solved the issue. Now `vagrant box list` works.

I never manually created those directories and I'm 90% sure all boxes are from Atlas, so it might have come from an old Vagrant version (< 2.x) that I used last year. The `ubuntu` directory that contained the `trusty64` directory was from May 2017. Maybe this helps in tracking down the combination of Vagrant version and box version that lead to this error.
